src/main/java/com/openshift/jenkins/plugins/freestyle/BaseStep.java

```diff
@@ -255,16 +260,18 @@ public boolean perform(ProcessBuilder pb)
                         new Thread(new Runnable() {
                             @Override
                             public void run() {
-                                byte buffer[] = new byte[1024];
-                                int count;
-                                try {
-                                    while ((count = output.read(buffer)) != -1) {
-                                        listener.getLogger().write(buffer, 0,
-                                                count);
+                                StringBuilder sb = new StringBuilder();
```

Comment:
StringBuilder is not synchronized, so, we may endup in log messages displayed in a different order from what they are produced.
Any objections in using StringBuffer instead ?

Reply:
@akram Thanks for the correction, I was not aware of the difference.